dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/message/DefaultProgramMessageService.java

```diff
@@ -267,37 +235,37 @@ public void validateQueryParameters(ProgramMessageQueryParams params) {
   @Override
   @Transactional(readOnly = true)
   public void validatePayload(ProgramMessage message) {
-    String violation = null;
+    List<String> violations = new ArrayList<>();
 
     ProgramMessageRecipients recipients = message.getRecipients();
 
     if (message.getText() == null) {
-      violation = "Message content must be provided";
+      violations.add("Message content must be provided");
     }
 
     if (message.getDeliveryChannels() == null || message.getDeliveryChannels().isEmpty()) {
-      violation = "Delivery channel must be specified";
+      violations.add("Delivery channel must be specified");
     }
 
     if (message.getEnrollment() == null && message.getEvent() == null) {
-      violation = "Enrollment or event must be specified";
+      violations.add("Enrollment or event must be specified");
     }
 
     if (recipients.getTrackedEntity() != null
         && trackedEntityService.getTrackedEntity(recipients.getTrackedEntity().getUid()) == null) {
-      violation = "Tracked entity does not exist";
+      violations.add("Tracked entity does not exist");
     }
 
     if (recipients.getOrganisationUnit() != null
         && organisationUnitService.getOrganisationUnit(recipients.getOrganisationUnit().getUid())
             == null) {
-      violation = "Organisation unit does not exist";
+      violations.add("Organisation unit does not exist");
     }
 
-    if (violation != null) {
-      log.info("Message validation failed: " + violation);
-
-      throw new IllegalQueryException(violation);
+    if (!violations.isEmpty()) {
+      String violationMessage = String.join(", ", violations);
+      log.info("Message validation failed: " + violationMessage);
```

Comment:
Why would you log this? How will this exception be handled?

When logging we should always think about who this log entry is for and what they should do about it. In this case I think a user should know that what they did was not valid and what they can do to fix it. Logging this will not help a user. It will just cause bigger log files operators have to store/seek through when looking for errors they should take action on.